### PR TITLE
Allow NPCs to read books

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Allow NPC to read books and learn on their own"```

#### Purpose of change
Allow NPC to read books and learn on their own.

#### Describe the solution
Ask NPC in dialogue to read a book from their inventory.
Currently only skill books - no use for morale books yet, and indeed there are some issues with the continuous reading thing that keeps restarting the activity, so the easiest solution was just for the NPC to stop reading when they gained a level, - whch isnt possible when reading morale books.

The reading code was very player-centric and UI based, I could've transplanted and hoisted all the reading code from avatar to player and modified it, but that wouldve been invasive and caused potential bugs to player reading.

This way , I've just made a minimal version of the player reading code in npc.cpp.

This is still manually prompted by talking to the NPC, but as with most of th eother activity code, the intention is to leave it open for future automation of NPC "jobs" and AI where they can choose to read in their downtime etc.

#### Describe alternatives you've considered
N/A

#### Additional context
N/A

### CODE REVIEW

**Comment by davidpwbrown** [comment]
Will fix the tests shortly.

**Comment by davidpwbrown** [comment]
Moved all the player/avatar-centric reading functions back to where they were to avoid potentially breaking anything, and to keep the tests happy. made a mini, stripped-down version of the reading function for NPC usage only, as they dont need all the other stuff like morale and reading to others ( yet )

File: src/npc.cpp [hunk 1]
```diff
@@ -799,6 +808,213 @@ void npc::starting_weapon( const npc_class_id &type )
...
+        if( skill_level != originalSkillLevel ) {
+            g->events().send<event_type::gains_skill_level>( getID(), skill, skill_level.level() );
+            if( display_messages ) {
+                add_msg( m_good, _( "%s increases their %s level." ), disp_name(), skill_name );
+                // NPC reads until they gain a level, then stop.
+                revert_after_activity();
+                return;
+            }
+        } else {
+            //skill_level == originalSkillLevel
```
Comment (Night-Pryanik) [inline src/npc.cpp]:
```suggestion
```

File: src/npc.cpp [hunk 2]
```diff
@@ -799,6 +808,213 @@ void npc::starting_weapon( const npc_class_id &type )
...
+    if( skill && get_skill_level( skill ) < reading->level &&
+        get_skill_level_object( skill ).can_train() ) {
+        SkillLevel &skill_level = get_skill_level_object( skill );
+        const int originalSkillLevel = skill_level.level();
+
+        // Calculate experience gained
+        /** @EFFECT_INT increases reading comprehension */
+        // Enhanced Memory Banks modestly boosts experience
+        int min_ex = std::max( 1, reading->time / 10 + get_int() / 4 );
+        int max_ex = reading->time /  5 + get_int() / 2 - originalSkillLevel;
```
Comment (Night-Pryanik) [inline src/npc.cpp]:
```suggestion
        int max_ex = reading->time / 5 + get_int() / 2 - originalSkillLevel;
```

**Comment by adamkad1** [comment]
npcs are no longer children you need to read to? nice!

**Pull request merged**